File: src/main/java/aiai/ai/beans/Snippet.java
/*
 AiAi, Copyright (C) 2017 - 2018, Serge Maslyukov

 This program is free software: you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation, either version 3 of the License, or
 (at your option) any later version.

 This program is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.

 You should have received a copy of the GNU General Public License
 along with this program.  If not, see <https://www.gnu.org/licenses/>.

 */
package aiai.ai.beans;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "AIAI_LP_SNIPPET")
@Data
public class Snippet implements Serializable {
    private static final long serialVersionUID = 4066977399166436522L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Version
    private Integer version;

    @Column(name = "NAME")
    public String name;

    @Column(name = "SNIPPET_TYPE")
    public String type;

    @Column(name = "SNIPPET_VERSION")
    public String snippetVersion;

    @Column(name = "FILENAME")
    public String filename;

    /**
     * programm code, i.e. pyphon, js,...
     */
    @Column(name = "CODE")
    public String code;

    @Column(name = "CHECKSUM")
    public String checksum;

    public String getSnippetCode() {
        return ""+ name + ':' + snippetVersion;
    }
}

File: apps/angular/src/app/models/Snippet.ts
// export class Snippet {
//     name: string;
//     version: string;
//     type: string;
//     environment: string;
//     params: string;
//     isSigned: boolean;
//     constructor(p: any) {
//         this.name = p.name
//         this.version = p.version
//         this.type = p.type
//         this.environment = p.environment
//         this.params = p.params
//         this.isSigned = p.isSigned
//     }
// }


export class Snippet {
    id: number;
    version: number;
    name: string;
    type: string;
    snippetVersion: string;
    filename: string;
    params ? : null;
    checksum: string;
    env: string;
    isSigned: boolean;
    reportMetrics: boolean;
    length: number;
    fileProvided: boolean;
    signed: boolean;
    snippetCode: string;
    constructor(p: any) {
        this.id = p.id
        this.version = p.version
        this.name = p.name
        this.type = p.type
        this.snippetVersion = p.snippetVersion
        this.filename = p.filename
        this.params = p.params
        this.checksum = p.checksum
        this.env = p.env
        this.isSigned = p.isSigned
        this.reportMetrics = p.reportMetrics
        this.length = p.length
        this.fileProvided = p.fileProvided
        this.signed = p.signed
        this.snippetCode = p.snippetCode
    }
}